docs/shell-completion.md

```diff
@@ -1,45 +1,108 @@
 # Enabling Shell Completion
 
-Generate the k0s completion script using the `k0s completion <shell_name>` command, for Bash, Zsh, fish, or PowerShell.
+## Introduction
 
-Sourcing the completion script in your shell enables k0s autocompletion.
+Shell completion enhances the user experience by providing auto-completion for
+commands in the terminal. K0s supports shell completion for the following
+shells:
 
-## Bash
+- `bash`, [GNU Bash](https://www.gnu.org/software/bash/)
+- `zsh`, [the Z-shell](https://www.zsh.org/)
+- `fish`, [the friendly interactive shell]((https://fishshell.com/))
+- `powershell`, [Microsoft PowerShell](https://learn.microsoft.com/powershell/)
 
-```shell
-echo 'source <(k0s completion bash)' >>~/.bashrc
+## General Usage
+
+To generate a completion script for your shell, use the following command: `k0s
+completion <shell_name>`. Sourcing the completion script in your shell enables
+k0s autocompletion.
+
+## bash
+
+One-shot usage: `source <(k0s completion bash)`.
+
+This is a recipe to load completions for each new shell. Adjust to your personal
+needs:
+
+```bash
+mkdir ~/.bash_completion.d
+k0s completion bash >~/.bash_completion.d/k0s
+
+cat <<'EOF' >~/.bashrc
```

Comment:
how about something like this?

```sh
grep -q .bash_completion.d .bashrc || cat <<'EOF'
```

Reply:
I tried to find some references about the idiomatic way of including user-managed bash shell completions, but I didn't really find anything. The `bash_completion.d` stuff seems to be used, but there doesn't seem to be "the way of doing it". That's why I tried to give an example with the caveat "adjust to your personal needs". Checking if the folder has already been set up in .bashrc or some other place is not a piece of cake. That's why I opted for not doing it.

If you think checking would be better, then I'd probably change this to some descriptive block where to put the snippet, similar to the examples for zsh.

 WDYT?